src/zig.rs

```diff
@@ -963,6 +963,43 @@ pub struct ZigWrapper {
     pub ranlib: PathBuf,
 }
 
+#[derive(Debug, Clone, Default, PartialEq)]
+struct TargetFlags {
+    pub target_cpu: String,
+    pub target_feature: String,
+}
+
+impl TargetFlags {
+    pub fn parse_from_encoded(encoded: &OsStr) -> Result<Self> {
+        let mut parsed = Self::default();
+
+        let f = rustflags::from_encoded(encoded);
+        for flag in f {
+            if let rustflags::Flag::Codegen { opt, value } = flag {
+                let key = opt.replace('-', "_");
+                match key.as_str() {
+                    "target_cpu" => {
+                        if let Some(value) = value {
+                            parsed.target_cpu = value;
+                        }
+                    }
+                    "target_feature" => {
```

Comment:
No point in replacing `-` with `_` here?

```suggestion
                match opt.as_str() {
                    "target-cpu" => {
                        if let Some(value) = value {
                            parsed.target_cpu = value;
                        }
                    }
                    "target-feature" => {
```

Reply:
Both -Ctarget_cpu and -Ctarget-cpu are recognizable by rustc. Also see https://github.com/rust-lang/rust/blob/e7da0fa62fc58f9d665a67bf2e7a6d6154d8d74c/compiler/rustc_session/src/options.rs#L340

Reply:
OK, thanks for the explanation.